src/main/java/com/google/devtools/build/lib/rules/python/PyRuntimeInfo.java

```diff
@@ -264,11 +278,22 @@ public PyRuntimeInfo constructor(
         Object coverageFilesUncast,
         String pythonVersion,
         String stubShebang,
+        Object bootstrapTemplateUncast,
         StarlarkThread thread)
         throws EvalException {
       String interpreterPath =
           interpreterPathUncast == NONE ? null : (String) interpreterPathUncast;
       Artifact interpreter = interpreterUncast == NONE ? null : (Artifact) interpreterUncast;
+      Artifact bootstrapTemplate = null;
+      if (bootstrapTemplateUncast != NONE) {
+        if (bootstrapTemplateUncast instanceof InputFileConfiguredTarget) {
```

Comment:
InputFileConfiguredTarget is definitely too low-level/concrete-implementation detail-y. TransitiveInfoCollection should be used instead for the equivalent of a Starlark Target.

However, per my other comments, don't accept a target here. Accept FileApi as an input (since that's what we're supposed to use in the "api interface" classes, from what i can tell), and then cast to Artifact in the implementation. (this all seems a bit roundabout, but it seems like that's what everything else is doing; and it looks like Artifact is, in practice, the only implementation of FileApi)